[Docs] improve NAT and manual peering documentation (#2827)
liqotech/liqo: docs/advanced/peering/inter-cluster-network.md

The unpeer process will automatically remove the Liqo Gateway from the tenant namespace.

> ## Setup the inter-cluster network via liqoctl network

Reviewer: ```suggestion
## Setup the inter-cluster network via 'liqoctl network' command
```

> ## Manual on cluster couple

Reviewer: Is it something like "option C"?
(previous subsections of type '##' started with "Option A" and "Option B".)

In any case, I don't understand what the title means.

Author: This title belongs to another page :)

Reviewer: Ops... got it.
In any case, the title is complex to understand to me. I would simply say "Manual".

Author: Agree that the title is not so clear. Trying to make them more descriptive

When you have access to both clusters, you can configure the inter-cluster network connectivity via the `liqoctl network` command.

> Note that when you use the `liqoctl network` command, the remote kubeconfig/context provided as argument, is the one of the cluster where we want to configure the server gateway of Wireguard tunnel.

Reviewer: ```suggestion
Note that when you use the `liqoctl network` command, the argument specifying the remote kubeconfig/context corresponds to the cluster that acts as server gateway for the Wireguard tunnel.
```

> The first step to configure networking, is initializing the network configuration:

Reviewer: Still missing what "initialize" means. What are we actually achieving with this command? Why is it needed?

Author: Fixed!
Initialize retrieves the network configurations of the clusters that needs to be connected, and creates all the resources to enable the IP remapping. The reason why this is a separate command is that this is a one shot action, but we are internally discussing whether it makes sense removing this command and making the `network init` implicit in the `network connect` command

Use the `liqoctl network connect --help` command to see all the available options.

## Manually setup the inter-cluster network on each cluster separately

> When you don't have access to both clusters, or you want to configure it in a declarative way, you can configure it by applying CRDs.

Reviewer: ```suggestion
When you do not have contemporary access to both clusters, or you would like to configure the inter-cluster network in a declarative way, you can configure this component by applying the proper CRDs on either clusters.
```

3. **Cluster client**: the other cluster defines a `GatewayClient` resource, which will configure a client that will connect to the gateway server exposed on the other cluster.
4. **Cluster client and server**: the cluster client and server need to exchange the public keys to allow secure communication.

> More info about the declarative setup of the network can he found [here](./peering-via-cr.md#declarative-network-configuration).

Reviewer: ```suggestion
More info about the declarative setup of the network can be found [here](./peering-via-cr.md#declarative-network-configuration).
```
